Client/Frontend/Browser/Search/InitialSearchEngines.swift

```diff
@@ -137,7 +156,7 @@ class InitialSearchEngines {
             .sorted { e, _ in e.id == priorityEngine }
     }
     
-    private func replaceOrInsert(engineId: SearchEngineID, customId: String) {
+    private func replaceOrInsert(engineId: SearchEngineID, customId: String?) {
         guard let engineIndex = engines.firstIndex(of: .init(id: engineId)) else {
```

Comment:
I believe no need to create an instance here, we can probably use 
`        guard let engineIndex = engines.firstIndex(where: { $0.id == engineId }) else {
`